### DISCUSSION TITLE
Relation prediction

### DISCUSSION THREAD

**Discussion by symd** [LukasZahradnik/PyNeuraLogic#39] [Q&A]
Do you have an example for predicting the relation between two nodes in a graph ? Or How this could be implemented with NL?

**Comment by GustikS** [comment]
Hi, good question, thanks. There is one example with knowledge-base completion in the backend https://github.com/GustikS/NeuraLogic/tree/master/Resources/datasets/relational/kbs/nations
but we are yet to bring it to the frontend. Basically, the point is you can just add the corresponding constants (nodes/relations) to the queries
```python
R.predict(C.node1,C.relation1,C.node2)[1.0]
...
```
and reflect that at the output of the template, too
```python
R.predict(V.N1,V.R,VN2) <= ...
```
- the rest of the template is up to you, there can be some simple embedding of the nodes and relations, or some GNN-like propagation etc. (or both combined, as in the example).
You can also treat the relations explicitly and not as terms/constants, should they e.g. lead to different underlying logic...

**Reply by symd** [reply]
Thank you for responding to my questions. I will try .
